projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php

```diff
@@ -1240,6 +1254,18 @@ public function get_related_post_data_for_post( $post_id, $position, $origin ) {
 				$post->ID
 			),
 		);
+
+		/**
+		 * Filter the array of related posts.
+		 *
+		 * @module related-posts
+		 *
+		 * @since 2.8.0
+		 *
+		 * @param array $results Array of related posts.
+		 * @param int $post_id Post ID of the post for which we are retrieving Related Posts.
+		 */
+		return apply_filters( 'jetpack_relatedposts_returned_results', $related_posts, $post_id );
```

Comment:
A filter already exists under this name, but I see both docblocks still have the full information.

Was it intentional to copy the filter wholesale? Should we use the typical notation that the filter is already documented?

Additionally, one docblock says the second param (`post_id`) is a string and the one says an int.

Reply:
Ooo, good spot. I didn't see that it had already been documented. @mkm119 I think we can replace this docblock with the [/** This is already documented...](https://github.com/Automattic/jetpack/blob/b02774b715654d884b82e0c1940f6d456cec2015/projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php#L1123) format, and update the parameter type in the other docblock, because I think it should be an `int`, but it's probably worth checking that.

Reply:
I was not aware of the filter is already documented notation. I will update this.

The string vs int thing is a bit confusing. The root method where the filter is being used has $post_id type as `int` in both places, so I believe that should be the right type. I will change the type to `int` [here](https://github.com/Automattic/jetpack/blob/ff4490a0ffe0e25b5f4f885e61126ea0cc3a5485/projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php#L864)

@kraftbj @pablinos Does that sound ok?

Reply:
Sounds good to me, thanks.

Reply:
Handled in #22697